src/protocol.h

```diff
@@ -136,25 +136,43 @@ class CInv
 public:
     CInv();
     CInv(int typeIn, const uint256& hashIn);
+    CInv(int typeIn, const uint256& hashIn, const uint256& hashAuxIn);
 
     ADD_SERIALIZE_METHODS;
 
     template <typename Stream, typename Operation>
     inline void SerializationOp(Stream& s, Operation ser_action)
     {
+        int nVersion = s.GetVersion();
         READWRITE(type);
         READWRITE(hash);
+        if (type == MSG_WTX && nVersion >= CINV_WTX_VERSION) {
+            // We've already checked above that nVersion >= CINV_WTX_VERSION.
+            READWRITE(hashAux);
+        } else if (type == MSG_TX && ser_action.ForRead()) {
+            // Ensure that this value is set consistently in memory for MSG_TX.
+            hashAux = uint256S("0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff");
+        }
     }
 
     friend bool operator<(const CInv& a, const CInv& b);
 
     std::string GetCommand() const;
+    std::vector<unsigned char> GetWideHash() const;
     std::string ToString() const;
 
     // TODO: make private (improves encapsulation)
 public:
     int type;
+    // The main hash. This is:
+    // - MSG_BLOCK: the block hash.
+    // - MSG_TX and MSG_WTX: the txid.
     uint256 hash;
+    // The auxiliary hash. This is:
+    // - MSG_BLOCK and MSG_TX: null (all-zeroes) and not parsed or serialized.
+    // - MSG_TX: legacy auth digest (all-ones) and not parsed or serialized.
```

Comment:
This is confusing; it seems to be saying that for `MSG_TX`, the auxiliary hash should be both all-zeroes and all-ones.

Reply:
At one point it was all-zeroes, then I changed it to all-ones because I needed that consistency, and I messed up the comments.